src/measure_theory/regular.lean

```diff
@@ -124,5 +198,444 @@ instance sigma_finite [opens_measurable_space α] [t2_space α] [sigma_compact_s
+    exact le_supr (λ s : {s // is_closed s ∧ s ⊆ V}, μ (s ∩ A)) ⟨F, F_closed, FV⟩ },
+  { exact lt_of_le_of_lt (measure_mono (inter_subset_right _ _)) h'A },
+  { apply_instance }
+end
+
+/-- In a metric space (or even a pseudo emetric space), an open set can be approximated from inside
+by closed sets. -/
+lemma inner_regular_of_pseudo_emetric_space {X : Type*}
+  [pseudo_emetric_space X] [measurable_space X] [borel_space X] (μ : measure X)
+  (U : set X) (U_open : is_open U) :
+  μ U ≤ ⨆ (F : set X) (hF : is_closed F) (FU : F ⊆ U), μ F :=
+begin
+  rcases U_open.exists_Union_is_closed with ⟨F, F_closed, F_subset, F_Union, F_mono⟩,
+  conv_lhs { rw ← F_Union },
+  rw measure_Union_eq_supr (λ n, (F_closed n).measurable_set) F_mono.directed_le,
+  simp only [supr_le_iff],
+  assume n,
+  simp_rw [supr_and', supr_subtype'],
+  exact le_supr (λ s : {s // is_closed s ∧ s ⊆ U}, μ s) ⟨F n, F_closed n, F_subset n⟩,
+end
+
+/-- Any finite measure on a metric space (or even a pseudo emetric space) is weakly regular. -/
+@[priority 100] -- see Note [lower instance priority]
+instance of_pseudo_emetric_space_of_finite_measure {X : Type*}
+  [pseudo_emetric_space X] [measurable_space X] [borel_space X] (μ : measure X) [finite_measure μ] :
+  weakly_regular μ :=
+weakly_regular_of_inner_regular_of_finite_measure μ $ inner_regular_of_pseudo_emetric_space μ
+
+end weakly_regular
+
+namespace regular
+
+/-- Any locally finite measure on a sigma compact locally compact metric space is regular. -/
+@[priority 100] -- see Note [lower instance priority]
+instance of_sigma_compact_space_of_locally_finite_measure {X : Type*}
+  [emetric_space X] [sigma_compact_space X] [locally_compact_space X]
+  [measurable_space X] [borel_space X] (μ : measure X)
+  [locally_finite_measure μ] : regular μ :=
+{ lt_top_of_is_compact := λ K hK, hK.measure_lt_top,
+  inner_regular :=
+  begin
+    /- Given an open set, it can be approximated from inside by closed sets thanks to
+    `inner_regular_of_pseudo_emetric_space`. Each such closed set `F` can be approximated by a
+    compact set `B n ∩ F`, using a compact exhaustion `B n` of the space. -/
+    assume U U_open,
+    refine (weakly_regular.inner_regular_of_pseudo_emetric_space μ U U_open).trans _,
+    simp only [supr_le_iff],
+    assume F F_closed FU,
+    have B : compact_exhaustion X := default _,
+    have : F = ⋃ n, B n ∩ F, by rw [← Union_inter, B.Union_eq, univ_inter],
+    rw [this, measure_Union_eq_supr
+      (λ n, (B.is_compact n).measurable_set.inter F_closed.measurable_set)],
+    { simp_rw [supr_le_iff, supr_and', supr_subtype'],
+      assume n,
+      exact le_supr (λ K : {K // is_compact K ∧ K ⊆ U}, μ K)
+        ⟨B n ∩ F, is_compact.inter_right (B.is_compact n) F_closed,
+          subset.trans (inter_subset_right _ _) FU⟩ },
+    { apply (monotone_of_monotone_nat (λ n, _)).directed_le,
+      exact inter_subset_inter_left _ (B.subset_succ n) },
+  end,
+  outer_regular :=
```

Comment:
I think this proof can be simplified by using `ennreal.exists_pos_sum_of_encodable`, abstracting away some of the calculations

Reply:
Yes, thanks!